### DISCUSSION TITLE
Moving from iTerm2

### DISCUSSION THREAD

**Discussion by karimlevallois** [kovidgoyal/kitty#5298] [Q&A]
Just trying to move over to Kitty and trying to mirror my setup.

In iTerm2 when I open the app I have a default window around 60c 20c in size, with font size of 12 and minimal titlebar (just the open/close button showing with background matching colour elsewhere for a seamless look). This always launches in the top left of my screen and I use this for basic terminal commands.

I then have a hotkey that launches nvim in full a fullscreen window with font size of 14 and adjusted line padding of 10.

What would be the easiest way to mirror this in Kitty?

Many thanks in advance!

**Comment by page-down** [comment]
Read the documentation and configure what you need. The macOS-related configuration options are under "OS specific tweaks".

https://sw.kovidgoyal.net/kitty/conf/

> ... always launches in the top left of my screen ...

This is currently not possible and requires other software or writing your own script to move the window.

> ... a fullscreen window with font size of 14 ...

In order to start full-screen neovim and change the font size, in addition to combining multiple actions using `combine` like below, you can also write custom kitten or use remote control for precise control.

```conf
map f1 combine : launch --type=os-window nvim : toggle_fullscreen : change_font_size current 14.0
```

Note that the font size is applied to the entire OS window, including tabs and windows.

> ... adjusted line padding ...

There is no spacing between each cell in kitty. You can try to modify the cell height.
However, the cell height cannot be changed on the fly, you need to run another kitty instance.

Maybe line spacing can be set individually for OS windows in the future, just like font size.

**Reply by karimlevallois** [reply]
Wow, thank you so much for such a detailed response. I did read the documentation online, but was trying to use a startup session and not getting exactly what I was after. This is perfect and the transition is so much smoother than iTerm2.

Just one last question if I may? In iTerm2 I use a different .zshrc file (https://michaelheap.com/custom-zshrc-iterm2/) for launching my nvim instance that drops neofetch. Can this be done inside my mapping?

**Reply by kovidgoyal** [reply]
launch sh -c "export ZSHDOTDIR=/path/to/folder/with/custom/.zshrc nvim"

Though I dont really understand why you need a zshrc file at all while running nvim.

**Reply by karimlevallois** [reply]
The .zshrc file is called upon when opening the integrated terminal and if I don’t use a custom version without the fetch element it causes issues when running Python files as it runs the Python command before the fetch and then loads the current buffer cwd. Or something along those lines and it errors out saying it cannot locate the file.

Thanks for the above, do I just add that inside the mapping?

map f1 combine : launch --type=os-window nvim : toggle_fullscreen : change_font_size current 14.0

**Reply by page-down** [reply]
Use `exe_search_path` to add your bin paths, such as `/usr/local/bin` or paths under `/opt`.
Otherwise, the executable may not be found when using `launch` under macOS.

https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.exe_search_path

> ... use a different .zshrc file ...

https://sw.kovidgoyal.net/kitty/launch/#cmdoption-launch-env

```conf
launch --env ZDOTDIR=/dir/path nvim
```

> ... it errors out saying it cannot locate the file.

Maybe put an `if` into zshrc and you won't need another one specifically for this case.

**Reply by karimlevallois** [reply]
Thanks for the suggestion, I’ve added a conditional inside the zshrc to fix this. Kitty works great as it has a different TERM name than neovim (xterm-256). Not so much iTerm2 or Terminal.app as they both also use ‘xterm-256’ as their names. Another tick for Kitty I guess!

Thank you both for your help, much appreciated!

**Reply by karimlevallois** [reply]
One last question if that’s OK? I use LunarVim (Neovim) config which has its own command under ~/.local/bin/lvim, so runs separately to nvim. I’d like to use this as my default editor so that I can use ‘open in’ etc., but it’s not working. I’ve tried adding it under exe_search_path and adding different combinations under the editor flag, but running out of ideas…?

**Reply by page-down** [reply]
> ... use this as my default editor ...

https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.editor

Make sure kitty is fully exited after configuration before opening it. Try to configure the absolute path, verify that the environment variables are configured.

I don't know how to guess `it's not working`, so you'll have to try it yourself.

**Reply by karimlevallois** [reply]
I meant that adding `editor ~/.local/bin/lvim` doesn’t seem to work as I get an error message when opening a file in Kitty. I also tried adding `exe_search_paths ~/.local/bin`  , but not luck. Wouldn’t it be easier to just add `export EDITOR=“~/.local/bin/lvim”` to my zshrc instead?

**Reply by page-down** [reply]
It depends on what exactly you are trying to accomplish.
Sorry I can't guess what it is. If you need help, you'll need to elaborate.
Describe each step exactly what you did, the results you got and the expectations.

For example: dragging a text file from Finder to kitty's Dock icon using the mouse, or clicking the mouse on the text file hyperlink, yet not opening it with the default editor. Or click on the menu bar and select "Preferences", however the kitty config file is not opened with the default editor. Or entered some command in the shell, but an error is reported.

> ... so that I can use ‘open in’ etc ...

What exactly is the "open in" function you are talking about?

> ... I get an error message ...

Since you didn't post what that error message is either, I can't help.

**Reply by karimlevallois** [reply]
Sorry, realise I may not be giving enough context. Basically I’ve set up a open-actions.conf exactly as the example on the below.

https://sw.kovidgoyal.net/kitty/open_actions/#scripting-the-opening-of-files-with-kitty-on-macos

I have tried opening various files, .txt, .py etc. via right click ‘open in Kitty’ and they open fine, but in what looks to be nano. I have tried updating the `editor` in my kitty.conf to `~/.local/bin/lvim` but I then get an error message when trying to open files. I have however tried changing the `editor` to nvim and that works fine, so it appears that I have the open-actions.conf sourced/set up correctly.

The below is an extract from the conf. which shows the action to open the `$EDITOR` which in my case I want to be lvim. I would also prefer the os-window to open fullscreen if that is possible?

```
# Open text files without fragments in the editor
protocol file
mime text/*
action launch --type=os-window $EDITOR $FILE_PATH
```

**Reply by page-down** [reply]
> ... updating the editor in my kitty.conf to ~/.local/bin/lvim ...

Try `editor lvim` or use the absolute path to lvim.

> ... prefer the os-window to open fullscreen ...

Try adding an action to switch to full screen under `action launch ...`.

**Reply by karimlevallois** [reply]
I’ve previously tried adding both `editor lvim` and `editor ~/.local/bin/lvim` but unfortunately I get a child operation error saying that the path doesn’t exist. I’ve also tried adding both variations in my zshrc under `EDITOR` and `VISUAL` but when I right click and open a document in Kitty a blank screen quickly flashes up and then Kitty crashes. I’ll keep troubleshooting and see if I can get it to work. I can use lvim via my mapping below, lvim only seems to launch when combined with `zsh` though, even after adding the paths in Kitty.

`map shift+ctrl+l combine : launch --type=os-window zsh ~/.local/bin/lvim : toggle_fullscreen : change_font_size current 14.0`

I’m probably being a bit silly, but I tried adding `toggle_fullscreen` in a few variations, but the window stayed the same.

action launch --type=os-window toggle_fullscreen $EDITOR $FILE_PATH

**Reply by karimlevallois** [reply]
This is the error message if I add `editor ~/.local/bin/lvim`. If I add just `editor lvim` nothing happens.

Failed to launch child: '~/.local/bin/lvim'
With error: No such file or directory
Press Enter to exit.

**Reply by karimlevallois** [reply]
I also have the following in my kitty.conf.

```
exe_search_path /opt
exe_search_path ~/.local/bin
```

**Reply by page-down** [reply]
> This is the error message ...
> ...
> ... No such file or directory

This has been addressed in a recent PR. Thanks for your detailed report.
When the next nightly is released, you can install the latest nightly and try "~/.local/bin/lvim" to see if it works.

> I also have the following ...

"~/.local/bin" is already there and does not need to be added again.
If `editor lvim` is still not working, you should probably try to configure an **absolute path** for now.

> ... but I tried adding toggle_fullscreen in a few variations ...

```conf
action launch ...
action toggle_fullscreen
```

**Reply by karimlevallois** [reply]
It works! Thank you so much for taking the time to run through this, much appreciated.

**Reply by karimlevallois** [reply]
Quick last question.... I have a mapping to open LunarVim (below), but for some reason it's not changing the cwd to that of the buffer. Can I add :cd %:p:h:gs,\\,/,:gs, ,\\ , in order to run the change in cwd as soon as it opens?

`map shift+ctrl+l combine : launch --type=os-window zsh ~/.local/bin/lvim : toggle_maximized : change_font_size current 14.0`

**Reply by karimlevallois** [reply]
No need to answer, I've just added the :cd inside my mapping within my LunarVim keybindings and that's fixed it.
